### PR TITLE
JESI-3384

### PR DESCRIPTION
**Pull request by axrs**
[JESI-3384](https://jesims.atlassian.net/browse/JESI-3384)

### CODE REVIEW

File: VERSION [hunk 1]
```diff
@@ -1 +1 @@
-5.2.2
```
Comment (axrs) [inline VERSION:1]: Not exactly breaking for us... But could be if others were using.

File: src/io/jesi/backpack/macros.cljc [hunk 2]
```diff
@@ -206,16 +207,16 @@ A single default expression can follow the clauses, and its value will be return
...
                      `(~sym [this# ~@args more#]
-                        (if (seq? more#)
-                          (~invoke-fn this# (list* ~@args more#))
-                          (~invoke-fn this# (concat ~args (list more#)))))
+                        (apply ~invoke-fn this# (concat ~args (if (seq? more#)
+                                                                more#
+                                                                (list more#)))))
                      `(~sym [this# ~@args more#]
-                        (~invoke-fn this# (list* ~@args more#))))))]
+                        (apply ~invoke-fn this# (concat ~args more#))))))]
```
Comment (axrs) [inline src/io/jesi/backpack/macros.cljc:214]: Tried to concat at compile time, but couldn't get it to work. It doesn't know what more is....

File: test/io/jesi/backpack/cache_test.cljc [hunk 3]
```diff
@@ -68,83 +68,47 @@
...
-          (is= 17 (cached-sum 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1))
-          (is= 18 (cached-sum 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1))
-          (is= 19 (cached-sum 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1))
-          (is= 20 (cached-sum 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1))
-          ;CLJS does not support arity over 21
-          #?(:clj
-             (do
-               (is= 21 (cached-sum 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1))
-               (is= 22 (cached-sum 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1))
-               (is= 23 (cached-sum 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1)))))))
```
Comment (axrs) [inline test/io/jesi/backpack/cache_test.cljc:110]: moved

File: test/io/jesi/backpack/macros_test.cljc [hunk 4]
```diff
@@ -222,3 +222,79 @@
...
+          (is= 3 (impl 1 1 1))
+          (is= 3 (apply impl [1 1 1]))
+          (is= 4 (impl 1 1 1 1))
+          (is= 4 (apply impl [1 1 1 1]))
+          (is= 19 (impl 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1))
+          (is= 19 (apply impl [1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1])))
+
+        (testing "invoking with unknown arity throws exception"
+          (try
+            (impl 1 1 1 1 1)
```
Comment (AndreTheHunter) [inline test/io/jesi/backpack/macros_test.cljc:269]: Why not use `(is `(thrown? ...`?

Comment (axrs) [reply test/io/jesi/backpack/macros_test.cljc:269]: forgot it exists. Updated

**Pull request merged**